src/zerocash/Address.cpp

```diff
@@ -47,23 +35,27 @@ bool PrivateAddress::operator!=(const PrivateAddress& rhs) const {
 	return !(*this == rhs);
 }
 
-const std::string PrivateAddress::getEncryptionSecretKey() const {
+const uint256& PrivateAddress::getEncryptionSecretKey() const {
     return this->sk_enc;
 }
 
-const std::vector<unsigned char>& PrivateAddress::getAddressSecret() const {
+const uint256& PrivateAddress::getAddressSecret() const {
     return this->a_sk;
 }
 
-PublicAddress::PublicAddress(): a_pk(ZC_A_PK_SIZE) {
-    this->pk_enc = "";
+PublicAddress::PublicAddress() {
+    
 }
 
-PublicAddress::PublicAddress(const std::vector<unsigned char>& a_pk, std::string& pk_enc) : a_pk(a_pk), pk_enc(pk_enc) {}
+PublicAddress::PublicAddress(const uint256& a_pk, uint256& pk_enc) : a_pk(a_pk), pk_enc(pk_enc) {}
 
-PublicAddress::PublicAddress(const PrivateAddress& addr_sk): a_pk(ZC_A_PK_SIZE) {
+PublicAddress::PublicAddress(const PrivateAddress& addr_sk) {
     std::vector<bool> a_sk_bool(ZC_A_SK_SIZE * 8);
```

Comment:
We should not need to use bit arrays. Everything in the note encryption is expressible in terms of bytes.